src/views/clusteroverview/overview/components/top-consumers-card/utils/DurationDropdown.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import DurationOption from './DurationOption';
+
+export type DurationDropdownProps = {
+  selectedDuration: any;
+  selectHandler: any;
+};
+
+const DurationDropdown: React.FC<DurationDropdownProps> = ({ selectedDuration, selectHandler }) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const onSelect = (event, durationOption: DurationOption) => {
+    selectHandler(durationOption);
+    setIsOpen(false);
+  };
+
+  return (
+    <Select
+      data-test-id="duration-select-dropdown"
+      isOpen={isOpen}
+      onToggle={() => setIsOpen(!isOpen)}
+      onSelect={onSelect}
+      selections={DurationOption.fromString(selectedDuration)?.getDropdownLabel()}
+      variant={SelectVariant.single}
+    >
+      {DurationOption.getAll().map((durationOption) => {
+        const dropdownLabel = durationOption.getDropdownLabel();
+        const durationValue = durationOption.getValue();
+
+        return (
+          <SelectOption
+            data-test-id={`${durationValue}`}
+            key={`${durationValue}`}
```

Comment:
can these two lines be replaced with `prop={propVal}`?

Reply:
Done. Good catch. I was using a combination of values and changed my mind, but obviously didn't clean up after myself.